server/model/monitor.js

```diff
@@ -1461,6 +1496,110 @@ class Monitor extends BeanModel {
         }
     }
 
+    /**
+     * Check expiry date based on RDAP
+     * @param {string} domain Domain to lookup
+     * @returns {void}
+     */
+    async checkDomainExpiryNotifications(domain) {
+        const notificationList = await Monitor.getNotificationList(this);
+        if (! notificationList.length > 0) {
+            // fail fast. If no notification is set, all the following checks can be skipped.
+            log.debug("monitor", "No notification, no need to send domain notification");
+            return;
+        }
+
+        let domainInfoBean = await R.findOne("domain_expiry_info", "domain = ?", [ domain ]);
+
+        let expiryDate;
+        if (domainInfoBean === null || getDaysBetween(new Date(domainInfoBean.lastCheck), new Date()) > 1) {
+            expiryDate = await getDomainExpiryDate(domain);
+
+            if (domainInfoBean === null) {
+                domainInfoBean = R.dispense("domain_expiry_info");
+                domainInfoBean.domain = domain;
+            } else if (new Date(expiryDate) > new Date(domainInfoBean.expiry)) {
+                await R.exec("DELETE FROM domain_expiry_notification_sent_history WHERE domain = ?", [
+                    domain
+                ]);
+            }
+
+            domainInfoBean.expiry = expiryDate;
+            domainInfoBean.lastCheck = new Date();
+            R.store(domainInfoBean);
+        } else {
+            log.debug("domain", `Domain expiry already checked recently for ${domain}, won't re-check.`);
+            expiryDate = new Date(domainInfoBean.expiry);
+        }
+
+        if (expiryDate === null) {
+            return;
+        }
+
+        const daysRemaining = getDaysRemaining(new Date(), expiryDate);
+        log.debug("domain", `${domain} expires in ${daysRemaining} days`);
```

Comment:
Since we also send a notification 

```suggestion
        log.warn("domain", `${domain} expires in ${daysRemaining} days`);
```